Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,358 @@
+/**
+ * Copyright Jacob Van Order 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Kitura
+import KituraNet
+import LoggerAPI
+import Credentials
+
+import SwiftyJSON
+
+import Foundation
+
+
+/// CredentialsTwitter is a plugin for the Credentials framework that authenticate using Twitter. This plugin uses
+/// Twitter's [“Sign in with Twitter”](https://dev.twitter.com/web/sign-in/implementing) use case. Roughly, there are
+/// three steps to this process: obtaining a request token, redirecting the user, and converting the request token to
+/// an access token. These three steps are marked down below. The goal of this process is create a UserProfile to give 
+/// back to Credentials.
+
+public class CredentialsTwitter: CredentialsPluginProtocol {
+
+    /// You must register an app at https://apps.twitter.com in order to get the consumer key.
+    let consumerKey: String
+    
+    /// You must register an app at https://apps.twitter.com in order to get the consumer secret.
+    let consumerSecret: String
+
+    private var oAuthToken: String = ""
+    private var oAuthTokenSecret: String = ""
+    private var oAuthTokenVerifier: String = ""
+
+    
+    /// Initialization method used to gather the consumer key and consumer secret used by Twitter.
+    ///
+    /// - Parameters:
+    ///   - consumerKey: The consumer key gathered from https://apps.twitter.com for your application.
+    ///   - consumerSecret: The consumer secret gathered from https://apps.twitter.com for your application.
+    public init(consumerKey: String,
+                consumerSecret: String) {
+        self.consumerKey = consumerKey
+        self.consumerSecret = consumerSecret
+    }
+
+    /// CredentialsPluginProtocol
+
+    /// The name of the plugin.
+    public var name: String {
+        return "Twitter"
+    }
+
+    /// User profile cache.
+    public var usersCache: NSCache<NSString, BaseCacheElement>?
+
+    /// An indication as to whether the plugin is redirecting or not.
+    public var redirecting: Bool {
+        return true
+    }
+
+    /// A delegate for `UserProfile` manipulation.
+    public var userProfileDelegate: UserProfileDelegate?
```

Comment:
It is in the FB implementation. I have to talk to their team about `weak` in Kitura because when I used `weak` versions of `self` in closures like  you normally would in iOS, objects that I thought would stick around were being let go and `self` was nowhere to be found. This is a larger issue that I need resolve by my talk at CocoaConf but not for this PR. Good eyes, though.